src/containerapp/azext_containerapp/containerapp_env_certificate_decorator.py

```diff
@@ -191,3 +193,36 @@ def list(self):
         managed_certs = self.get_managed_certificates(certificate_name)
         private_certs = self.get_private_certificates(certificate_name)
         return managed_certs + private_certs
+
+class ContainerappEnvCertificatePreviweUploadDecorator(ContainerappEnvCertificateUploadDecorator):
+    def validate_arguments(self):
+        # validate arguments
+        if self.get_argument_certificate_file() and self.get_argument_certificate_key_vault_url():
+            raise ValidationError("Cannot use certificate file/password with certificate identity/keyvaulturl at the same time")
+        if (not self.get_argument_certificate_file()) and (not self.get_argument_certificate_key_vault_url()):
+            raise ValidationError("Either certificate file/password or certificate identity/keyvaulturl should be set when hostName is set")
+
+    def set_up_certificate_from_key_vault(self):
+        if self.get_argument_certificate_key_vault_url():
+            identity = self.get_argument_certificate_identity()
+            if not identity:
+                identity = "system"
+            if identity.lower() != "system":
+                subscription_id = get_subscription_id(self.cmd.cli_ctx)
+                identity = _ensure_identity_resource_id(subscription_id, self.get_argument_resource_group_name(), identity)
+            self.certificate["properties"]["certificateKeyVaultProperties"] = {
+                "keyVaultUrl": self.get_argument_certificate_key_vault_url(),
+                "identity": identity
+            }
+            # used for autogenrate cert name
+            super().set_argument_thumbprint("cert-kv")
+
+    def construct_payload(self):
+        self.set_up_certificate_from_key_vault()
+        super().construct_payload()
+
+    def get_argument_certificate_identity(self):
+        return self.get_param("certificate_identity")
+
+    def get_argument_certificate_key_vault_url(self):
+        return self.get_param("certificate_key_vault_url")
```

Comment:
At the end of file should add one blank line:
PEP 8: W292 no newline at end of file
![image](https://github.com/Azure/azure-cli-extensions/assets/46143499/3e3cc164-fa12-4f22-82e1-6a6bd17ed8be)